src/features/transactions/swapEventSubscriber.ts

```diff
@@ -126,23 +20,62 @@ export default function subscribeToSwapEvents(params: {
   dispatch: Dispatch;
 }) {
   const { swapContract, account, dispatch, chainId } = params;
-  swapContract.on("Swap", async (...data) => {
-    const transactions = store.getState().transactions;
-    const { nonce, signerWallet, transactionHash, protocol, transaction } =
-      mapSwapEvent(data, chainId, account, transactions);
 
-    // If we don't have a 'transaction', we don't already know about this swap
-    // and therefore don't need to update the UI.
-    if (!transaction) return;
+  const _account = account.toLowerCase();
+  const wrapperAddress = Wrapper.getAddress(chainId).toLowerCase();
```

Comment:
Yes, potentially - this code is from a different branch though (it just looks new because I moved it from another file) - I'd rather keep the scope of this PR limited to fetching transactions so if this is a significant problem we should open a new issue for it.

(It's not trivial to change, and AFAIK wrapper is deployed as part of our standard set of contracts?).

Feel free to open a separate issue if you want

Reply:
Okay did not understood that was old code moved :)